src/api/endpoint/api/publish.js

```diff
@@ -149,6 +157,10 @@ export function publishPackage(storage: IStorageHandler, config: Config) {
         storage.changePackage(packageName, metadata, req.params.revision, function(error) {
           afterChange(error, API_MESSAGE.PKG_CHANGED, metadata);
         });
+      } else if (isObject(metadata._attachments) && !Object.keys(metadata._attachments).length) {
+        storage.changePackage(packageName, metadata, 'deprecate', function(error) {
```

Comment:
Please create `const DEPRECATE = 'deprecate';` so you can use it here and also on `local-storage`.

Reply:
https://github.com/honzahommer/verdaccio/commit/b785f4545a615512feec4bb80d5875f47c4e9b5a

Reply:
> I'd rename the branch, it seems to be the issue, or you fork, I'm not sure, we need to make it run the checks otherwise we cannot merge when we finish the CR.

I've merged master into my branch.